### PR TITLE
Add more translations to dashboard and introduce i18n test wrapper

### PR DESCRIPTION
**Pull request by scottcarol**
Follow-up to #4866

Context: Although the majority of the dashboard text is already wrapped in `<Trans>` translation components, there are still some forms and help text that need to be wrapped.

This PR adds those translations and introduces an i18n wrapper into the front-end tests so that test files can have access to the translations if needed. 

This PR should not change any UI when viewing in English.

Note: The front-end tests are currently only testing the language `en` but this can be expanded in the future to all existing languages we support.

### CODE REVIEW

File: web/app/js/components/BreadcrumbHeader.jsx [hunk 1]
```diff
@@ -14,6 +14,7 @@ const routeToCrumbTitle = {
   top: 'Top',
   routes: 'Top Routes',
   community: 'Community',
+  gateways: 'Gateways',
```
Comment (scottcarol) [inline web/app/js/components/BreadcrumbHeader.jsx:17]: This is a bonus fix unrelated to translations - I noticed that the header title when viewing `/gateways` was not properly capitalized

File: web/app/js/locales/en/messages.json [hunk 2]
```diff
@@ -1,10 +1,24 @@
...
+  "Control plane": "Control plane",
   "Control plane components": "Control plane components",
+  "Current {cmdNameDisplay} query": "Current {cmdNameDisplay} query",
   "Data plane proxies": "Data plane proxies",
+  "LinkerdIsUpToDateMsg": "Linkerd is up to date.",
+  "New service profile": "New service profile",
   "NoDataToDisplayMsg": "No data to display",
+  "Pod status: {0}": "Pod status: {0}",
+  "Service mesh details": "Service mesh details",
+  "Update Now": "Update Now",
```
Comment (alpeb) [inline web/app/js/locales/en/messages.json:15]: I think using the English phrase as key makes things easier, but it appears the original convention was to introduce camel-case keys such as `buttonCancel`, but this update isn't always following that convention?

Comment (scottcarol) [reply web/app/js/locales/en/messages.json:15]: Good question, the camelCase keys such as `buttonCancel` are intended to be if the text is tied to a specific UI component, such as a button, form, etc., but there are instances of text throughout the dashboard that are just headlines/titles that are not tied to a specific display component or style, so I kept those as English phrases.

File: web/app/js/locales/es/messages.json [hunk 3]
```diff
@@ -1,10 +1,24 @@
 {
   "404Msg": "Página no encontrada.",
+  "A new version ({versionText}) is available.": "Una nueva version ({versionText}) esta disponible",
```
Comment (alpeb) [inline web/app/js/locales/es/messages.json:3]: `Una nueva versión ({versionText}) está disponible`

File: web/app/js/locales/es/messages.json [hunk 4]
```diff
@@ -1,10 +1,24 @@
...
   "All namespaces have a {productName} install.": "Todos los namespaces tienen una instalación de {productName}.",
+  "Control plane": "Plano de control",
   "Control plane components": "Componentes del plano de control",
+  "Current {cmdNameDisplay} query": "Consulta {cmdNameDisplay} actual",
   "Data plane proxies": "Proxies del plano de datos",
+  "LinkerdIsUpToDateMsg": "Linkerd está actualizado.",
+  "New service profile": "Nuevo service profile",
   "NoDataToDisplayMsg": "No hay datos que mostrar",
+  "Pod status: {0}": "Estado pod: {0}",
+  "Service mesh details": "Detalles de la malla de servicio",
```
Comment (alpeb) [inline web/app/js/locales/es/messages.json:14]: `Detalles de la malla de servicios`

File: web/app/js/locales/es/messages.json [hunk 5]
```diff
@@ -50,16 +64,33 @@
...
   "componentsMsg": "Componentes",
+  "createNewProfileMsg": "También puede crear un nuevo perfil",
   "formAuthority": "Authority",
+  "formAuthorityHelpText": "Mostrar solicitudes con este :authority",
+  "formCreateServiceProfileHelpText": "Para crear un service profile, descargue un perfil y luego aplíquelo con `kubectl apply`.",
   "formHTTPMethod": "Método HTTP",
+  "formHTTPMethodHelpText": "Mostrar solicitudes con este método HTTP",
   "formHideFilters": "Ocultar filtros",
   "formMaxRPS": "Max PPS",
+  "formMaxRPSHelpText {defaultMaxRps}": "Máximo de solicitudes por segundo para tap. Defecto {defaultMaxRps}",
```
Comment (alpeb) [inline web/app/js/locales/es/messages.json:75]: `Máximo de solicitudes por segundo para tap. Por defecto {defaultMaxRps}`

File: web/app/js/locales/es/messages.json [hunk 6]
```diff
@@ -50,16 +64,33 @@
...
   "formNamespace": "Namespace",
+  "formNamespaceErrorText": "El namespace debe constar de caracteres alfanuméricos en minúscula o '-' y debe comenzar y terminar con un carácter alfanumérico",
+  "formNamespaceHelpText": "Namespace para consultar",
+  "formNoNamedRouteTrafficFound": "No se encontró tráfico de ruta con nombre. Esto podría deberse a que el servicio no está recibiendo tráfico o porque no hay ningún perfil de servicio configurado. ¿El servicio tiene un service profile?",
```
Comment (alpeb) [inline web/app/js/locales/es/messages.json:79]: ... `no hay ningún service profile configurado.`

File: web/app/js/locales/es/messages.json [hunk 7]
```diff
@@ -50,16 +64,33 @@
...
   "formPath": "Ruta",
+  "formPathHelpText": "Mostrar solicitudes con rutas que comienzan con este prefijo",
+  "formResource": "Resource",
```
Comment (alpeb) [inline web/app/js/locales/es/messages.json:82]: `Recurso`

File: web/app/js/locales/es/messages.json [hunk 8]
```diff
@@ -50,16 +64,33 @@
...
+  "formResourceHelpText": "Resource para consultar",
```
Comment (alpeb) [inline web/app/js/locales/es/messages.json:83]: `Recurso para consultar`

File: web/app/js/locales/es/messages.json [hunk 9]
```diff
@@ -50,16 +64,33 @@
...
   "formScheme": "Esquema",
+  "formSchemeHelpText": "Mostrar solicitudes con este esquema",
+  "formServiceNameErrorText": "El nombre del servicio debe constar de caracteres alfanuméricos en minúscula o '-' comenzar con un carácter alfabético y terminar con un carácter alfanumérico",
   "formShowFilters": "Mostrar más filtros",
   "formToNamespace": "Al Namespace",
+  "formToNamespaceHelpText": "Namespace del recurso de destino",
   "formToResource": "Al Recurso",
+  "formToResourceHelpText": "Recurso objetivo",
```
Comment (alpeb) [inline web/app/js/locales/es/messages.json:91]: `Recurso destino`

**Review by alpeb — COMMENTED** [review]
Thanks @scottcarol !

Here's a list of things that I found haven't been translated yet:

- `Running Linkerd db709fd2 (git)` at the bottom of the sidebar
- In the Control Plane section:
  - Title `Control Plane`
  - `The service mesh was successfully installed!`
- When looking at a pod:
  - `LIVE CALLS` and `ROUTE METRICS`
  - `FROM` and `TO`
  - In the dependency graph, when a pod is unmeshed it's titled with `Unmeshed`
  - In the popups, the table headers `Source` and `Destination`
- Tap section:
  - `Resource`, `To Resource`  dropdown titles
  - The `Request Details` popup contents haven't been translated
- Top section:
  - `Resource` dropdown title
- `Top Routes` section title
- `Community` section title

**Comment by scottcarol** [comment]
@alpeb Thanks for the review! I just pushed changes that should fill in the missing translations you listed in your comment.

File: web/app/js/locales/es/messages.json [hunk 10]
```diff
@@ -29,7 +30,9 @@
   "columnTitleClusterName": "Nombre del Clúster",
   "columnTitleCount": "Total",
   "columnTitleDeployment": "Deployment",
+  "columnTitleDestination": "Destino",
   "columnTitleDirection": "Dirección",
+  "columnTitleFrom": "DE",
```
Comment (alpeb) [inline web/app/js/locales/es/messages.json:35]: `DESDE`

File: web/app/js/locales/es/messages.json [hunk 11]
```diff
@@ -44,6 +47,7 @@
   "columnTitleMethod": "Método",
   "columnTitleName": "Nombre",
   "columnTitleNamespace": "Namespace",
+  "columnTitleNoTraffic": "No Tráfico",
```
Comment (alpeb) [inline web/app/js/locales/es/messages.json:50]: `No Hay Tráfico`

File: web/app/js/locales/es/messages.json [hunk 12]
```diff
@@ -57,22 +61,32 @@
   "columnTitleRoute": "Ruta",
   "columnTitleSecured": "Asegurado",
   "columnTitleService": "Servicio",
+  "columnTitleSource": "Fuente",
   "columnTitleSuccessRate": "Tasa de éxito",
   "columnTitleTap": "Tap",
+  "columnTitleTo": "Á",
```
Comment (alpeb) [inline web/app/js/locales/es/messages.json:67]: `HACIA`

File: web/app/js/locales/es/messages.json [hunk 13]
```diff
@@ -57,22 +61,32 @@
...
+  "columnTitleUnmeshed": "No en la malla de servicios",
```
Comment (alpeb) [inline web/app/js/locales/es/messages.json:68]: `Ausente en la malla de servicios`

File: web/app/js/locales/es/messages.json [hunk 14]
```diff
@@ -129,9 +148,16 @@
   "tableTitleLeafServices": "Servicios Hoja",
   "tableTitleOutbound": "Salida",
   "tableTitlePods": "Pods",
+  "tableTitleRequestDetails": "Pedir Detalles",
```
Comment (alpeb) [inline web/app/js/locales/es/messages.json:151]: `Detalle Solicitudes`

**Review by alpeb — COMMENTED** [review]
Thanks @scottcarol , I've added some more comments :slightly_smiling_face:

**Comment by scottcarol** [comment]
@alpeb Thanks, just pushed changes addressing your feedback.

**Review by alpeb — APPROVED** [review]
Perfect! :+1:

**Review by cpretzer — APPROVED** [review]
nice work @scottcarol 

:shipit:

**Pull request merged**